Associative-Containers/exercise2.cpp

```diff
@@ -49,32 +42,31 @@ ErrorCode process(std::string input, double *out)
         {'%', [](double a, double b){ return static_cast<int>(a) % static_cast<int>(b); }},
         {'^', [](double a, double b){ return std::pow(a, b); }},
         {'$', [](double a, double b){return pow(a,1 / b);}},
-        {'!', [](double a, double b)
-            {   
-                double factorial;
-                for(int i = 2;i<static_cast<int>(a);++i){
-                    factorial*=i;
-                }
-                return factorial;
-            }
-        }
+        // {'!', [](double a, double )
+        //     {   
+        //         double factorial =1;
+        //         for(int i = 2;i<=static_cast<int>(a);++i){
+        //             factorial*=i;
+        //         }
+        //         return factorial;
+        //     }
+        // }
```

Comment:
The factorial algorithm looks fine and good that you chose the iterative version, but the challenge will be to handle the single operator in the further part of the function, because you need to handle operation with single argument differently.